app/modules/net_info.c

```diff
@@ -0,0 +1,201 @@
+/* 
+ * http://blog.mclemon.io/esp8266-contributing-to-the-nodemcu-ecosystem
+ * https://github.com/smcl/nodemcu-firmware/blob/cc04aaf92c1c076c30ef0b0eee43b3f924137440/app/modules/test.c
+ *
+ * extracted ping function
+ * adoptet jan 2017 for commit to nodeMCU by Wolfgang Rosner
+ */
+
+
+#include "module.h"
+#include "lauxlib.h"
+#include "platform.h"
+
+#include "c_stdlib.h"
+
+#include "lwip/ip_addr.h"
+#include "espconn.h"
+#include "lwip/dns.h" 
+#include "lwip/app/ping.h"
+#include "lwip/raw.h"
+#include "c_stdio.h"
+
+
+#include "task/task.h"
+
+
+
+// https://github.com/nodemcu/nodemcu-firmware/wiki/[DRAFT]-How-to-write-a-C-module#debug-and-error-messages
+
+#define NET_INFO_DEBUG_ON
+
+#define log_prefix " ### DEBUG: net_info ### :  "
+
+#if defined(DEVELOP_VERSION)
+  #define NET_INFO_DEBUG_ON
+#endif
+#if defined(NET_INFO_DEBUG_ON)
+  #define NET_INFO_DEBUG(format, ...) dbg_printf("%s"format"", log_prefix, ##__VA_ARGS__)
+#else
+  #define NET_INFO_DEBUG(...)
+#endif
+#if defined(NODE_ERROR)
+  #define NET_INFO_ERR(format, ...) NODE_ERR("%s"format"\n", log_prefix, ##__VA_ARGS__)
+#else
+  #define NET_INFO_ERR(...)
+#endif
+
+
+#define NET_INFO_PRIORITY_OUTPUT   TASK_PRIORITY_MEDIUM
+#define NET_INFO_PRIORITY_ERROR    TASK_PRIORITY_MEDIUM
+
+// these statics should go away for reentrancy and maybe other reasons!
+static int ping_callback_ref; 
+static int ping_host_count;
+static ip_addr_t ping_host_ip;
+struct ping_option *ping_opt = NULL;
+
+void ping_received(void *arg, void *data) {
+    // this would require change of the interface 
+    // struct ping_msg *pingmsg = (struct ping_msg*)arg;
+    // struct ping_option *pingopt = pingmsg->ping_opt;
+    struct ping_option *pingopt = (struct ping_option*)arg;
+    struct ping_msg *pingmsg = pingopt->parent_msg ; 
+
+    struct ping_resp *pingresp = (struct ping_resp*)data;
+
+    char ipaddrstr[16];
+    ip_addr_t source_ip;
+    
+    source_ip.addr = pingopt->ip;
+    ipaddr_ntoa_r(&source_ip, ipaddrstr, sizeof(ipaddrstr));
+
+    if (ping_callback_ref != LUA_NOREF) {
+      lua_State *L = lua_getstate();
+      lua_rawgeti(L, LUA_REGISTRYINDEX, ping_callback_ref);
+      lua_pushinteger(L, pingresp->bytes);
+      lua_pushstring(L, ipaddrstr);
+      lua_pushinteger(L, pingresp->seqno);
+      lua_pushinteger(L, pingresp->ttl);
+      lua_pushinteger(L, pingresp->resp_time);
+      lua_call(L, 5, 0);
+    } else {
+      c_printf("%d bytes from %s, icmp_seq=%d ttl=%d time=%dms\n",
```

Comment:
While I appreciate this as a form of debugging, I don't think it's an appropriate API for ping with no callback to spew to the console.  I think the callback should be mandatory and you can provide an example in the docs of a Lua function that prints to the console.